@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -101,6 +109,41 @@ class AttachmentCache {
     >();
 }
 
+class MemoCall extends CallExpr implements OperatorDescriptor {
+    readonly deps: string[];
+    readonly cache = new Map<string, Value>();
+    cachedValue?: Value;
+
+    constructor(expr: Expr) {
+        super("memo", [expr]);
+        this.deps = Array.from(expr.dependencies().properties);
+        this.descriptor = this;
+    }
+
+    call(context: ExprEvaluatorContext): Value {
+        const expr = this.args[0];
+
+        let changed = false;
+
+        for (const d of this.deps) {
+            const newValue = context.env.lookup(d);
+            if (!changed && newValue !== this.cache.get(d)) {
```

Comment:
Know that's microoptimization, but `this.deps` is array.
So `this.cache` can be array too.
Only change indexing of this loop to `i`. We save some unnecessary lookups and unnecessary maintenance of small Maps.

Reply:
done